apps/heureka/src/components/Services/ServicesList/index.tsx

```diff
@@ -25,36 +25,46 @@ type ServiceListProps = {
 }
 
 export const ServicesList = ({
+  defaultSelectService,
   services,
   loading,
   error,
   currentPage,
   totalNumberOfPages,
   goToPage,
 }: ServiceListProps) => {
-  const dispatch = useDispatch()
+  const navigate = useNavigate()
   const [selectedOverviewService, setSelectedOverviewService] = useState<ServiceType | null>(null)
 
-  const handleServiceOverviewOpen = useCallback((service: ServiceType) => {
+  const openServiceOverviewPanel = useCallback((service: ServiceType) => {
     setSelectedOverviewService((prev) => (prev?.id === service.id ? null : service))
+    navigate({
+      to: "/services",
+      search: {
+        service: service.name,
+      },
+    })
   }, [])
 
-  const handlePanelClose = useCallback(() => {
+  const closeServiceOverviewPanel = useCallback(() => {
     setSelectedOverviewService(null)
+    navigate({
+      to: "/services",
+    })
   }, [])
 
-  const goToServiceDetails = useCallback((service: ServiceType) => {
-    dispatch({
-      type: ActionType.SelectView,
-      payload: {
-        viewId: UserView.ServiceDetails,
-        params: {
-          service,
-        },
-      },
+  const goToServiceDetailsPage = useCallback((service: ServiceType) => {
+    navigate({
+      to: "/services/$service",
+      params: { service: service.name },
     })
   }, [])
 
+  // Keep selectedOverviewService in sync
+  useEffect(() => {
```

Comment:
could you clarify of usage of this useEffect

Reply:
This is to keep internal state in sync whenever `sevices` array or the default selected service change. Might not be needed in future when we convert this code to React19.

Reply:
We agreed on focus only on Tasntack Router part on this PR so this useEffect will handle separated.